src/autoscaler/api/custom_metrics_cred_helper/custom_metric_cred_helper.go

```diff
@@ -12,6 +12,28 @@ const (
 	MaxRetry = 5
 )
 
+type Credentials interface {
+	CreateCredential(appId string, userProvidedCredential *models.Credential, policyDB db.PolicyDB, maxRetry int) (*models.Credential, error)
```

Comment:
We should remove `policyDB db.PolicyDB` from the interface methods and add these to the constructor